os/wqueue/work_signal.c

```diff
@@ -92,25 +92,29 @@ int work_signal(int qid)
 
 #if defined(CONFIG_SCHED_USRWORK) && !defined(__KERNEL__)
 	if (qid == USRWORK) {
-		pid = g_usrwork.worker[0].pid;
+		struct wqueue_s *usr_work;
+		usr_work = get_usrwork();
+		pid = usr_work->worker[0].pid;
 	} else
 #endif
 #ifdef CONFIG_SCHED_HPWORK
 	if (qid == HPWORK) {
-		pid = g_hpwork.worker[0].pid;
+		struct hp_wqueue_s *hp_work;
+		hp_work = get_hpwork();
+		pid = hp_work->worker[0].pid;
 	} else
 #endif
 #ifdef CONFIG_SCHED_LPWORK
 	if (qid == LPWORK) {
 		int wndx;
 		int i;
-
+		struct lp_wqueue_s *lp_work;
 		/* Find an IDLE worker thread */
-
+		lp_work = get_lpwork();
```

Comment:
How about do you initialize the pointer as NULL. and after assign any pointer value.
Or when you declare the pointer variable, you can assign any poiniter value. 

Anyway, like this implementation can get some warnings from svace or coverity.

Reply:
Okay,  I initialized it to NULL